src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -119,50 +106,49 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
                             }
                             else
                             {
-                                var printPackages = true;
-
-                                // Handle --outdated
-                                if (listPackageArgs.IncludeOutdated)
+                                if (!listPackageArgs.IsOffline)  // --offline -- no server lookups
                                 {
+                                    await GetRegistrationMetadataAsync(packages, listPackageArgs);
                                     await AddLatestVersionsAsync(packages, listPackageArgs);
+                                }
 
-                                    printPackages = await FilterOutdatedPackagesAsync(packages);
-
-                                    // If after filtering, all packages were found up to date, inform the user
-                                    // and do not print anything
+                                // Filter packages for dedicated reports, inform user if none
+                                var printPackages = true;
+                                if (listPackageArgs.OutdatedReport)  // --outdated report
+                                {
+                                    printPackages = FilterOutdatedPackages(packages);
                                     if (!printPackages)
                                     {
                                         Console.WriteLine(string.Format(Strings.ListPkg_NoUpdatesForProject, projectName));
                                     }
                                 }
-                                // Handle --deprecated
-                                else if (listPackageArgs.IncludeDeprecated)
+                                else if (listPackageArgs.DeprecatedReport)  // --deprecated report
                                 {
-                                    await GetDeprecationInfoAsync(packages, listPackageArgs);
-
-                                    printPackages = await FilterDeprecatedPackagesAsync(packages);
-
-                                    // If after filtering, no packages were found to be deprecated, inform the user
-                                    // and do not print anything
+                                    printPackages = FilterDeprecatedPackages(packages);
                                     if (!printPackages)
                                     {
                                         Console.WriteLine(string.Format(Strings.ListPkg_NoDeprecatedPackagesForProject, projectName));
                                     }
                                 }
+                                else if (listPackageArgs.VulnerableReport)  // --vulnerable report
```

Comment:
I don't think the comment adds anything that the variable being checked doesn't say